crates/core/tedge_api/src/entity_store.rs

```diff
@@ -496,7 +496,7 @@ impl EntityStore {
             EntityType::Service => message
                 .parent
                 .clone()
-                .or_else(|| topic_id.default_parent_identifier())
+                .or_else(|| topic_id.default_service_parent_identifier())
```

Comment:
My first attempt as been to rewrite [TopicId::default_parent_identifier](https://github.com/thin-edge/thin-edge.io/blob/main/crates/core/tedge_api/src/mqtt_topics.rs#L403) along the following lines:

```rust
    /// Returns the topic identifier of the parent of an entity
    pub fn default_parent_identifier(&self) -> Option<Self> {
        match self.0.split('/').collect::<Vec<&str>>()[..] {
            ["device", "main", "", ""] => None,
            ["device", _, "", ""] => Some("main"),
            ["device", parent_id, "service", _] => Some(parent_id),
            _ => None,
        }
        .map(|parent_id| EntityTopicId(format!("device/{parent_id}//")))
    }
```

Indeed the current behavior of this function is weird (the parent of a child device is the child itself) and doesn't match is doc comment.

However, my proposal matches the doc comment but makes fail 35 unit tests!

@albinsuresh any idea on why this is not the correct way to fix this issue?

Reply:
Yeah, the current impl of `default_parent_identifier` seems wrong as it implies that the parent of `device/child1//` is `child1` which doesn't make any sense. That logic only applies to `service` entities following the default topic scheme. So, the new function that you've added is correct and named appropriately. I'd drop the old one in favour of the new one.

Reply:
Regarding the test failures (I looked at some, not all), it's coming from the auto-registration logic that tries to register `device/child1//` entity when a message like `te/device/child1///m/temp` is received. That auto-registration logic relies on the same `default_parent_identifier` function to extract the target entity topic id from the incoming topic. In this case, theyre' not looking for the parent, but the target entity itself. So, the current `default_parent_identifier` function can be renamed to `default_target_entity_identifier` to better reflect that logic, and the newly added function can be used to detect the "parent" of a "service".

Reply:
>  In this case, they're not looking for the "parent", but the "source" entity itself. 

That's really confusing but is aligned with the code (returning self for child devices and the parent for a service).

=> I will fix the name and the doc comment.

Reply:
> => I will fix the name and the doc comment.

Done: https://github.com/thin-edge/thin-edge.io/pull/3134/commits/c24e633d49047801a3d0e7af77a52ee7573978d4